lib/addDefaults.js

```diff
@@ -190,6 +191,59 @@ function lightDefaults(gltf) {
     }
 }
 
+function checkIfFloatVec4(value) {
+    if (defined(value) && value.length === 4) {
+        if (typeof(value[0]) === 'number' &&
+            typeof(value[1]) === 'number' &&
+            typeof(value[2]) === 'number' &&
+            typeof(value[3]) === 'number') return value;
+    }
+    else return [0, 0, 0, 1];
+}
+
+function guessExtensionTechnique(material) {
+    // Replaces the material technique, extension, and values with a close equivalent from KHR_materials_common.
+    // If no equivalent can be found, defaults to blinn for specular materials and lambert otherwise
+    var technique = 'LAMBERT';
+    var useBlinn = false;
+    var values = defined(material.values) ? material.values : {};
+
+    var ambient = checkIfFloatVec4(values.ambient);
+    var diffuse = typeof(values.diffuse) === 'string' ? values.diffuse : checkIfFloatVec4(values.diffuse);
+    var doubleSided = typeof(values.doubleSided) === 'boolean' ? values.doubleSided : false;
```

Comment:
Would a gltf material actually have a `doubleSided` property? Also this could just be made a bit cleaner as `defaultValue(values.doubleSided, false)`